src/main/java/step5/domain/RacingGame.java

```diff
@@ -0,0 +1,71 @@
+package step5.domain;
+
+import step5.dto.InputDto;
+import step5.view.ResultView;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class RacingGame {
+    private final List<Car> cars;
+    private InputDto inputDto;
+
+    // InputDto 랑 RacingGame 이 의존관계를 맺고 있어서 좋지 않은 거 같기도 한 느낌 입니다 ㅠ (이거 떄매 테스트 하기가 힘든 거 같습니담..)
+    // InputDto 를 없애고 Refactor 할 수 있을 거 같은데 당장은 떠오르지 않아서 일단.. 리뷰 받고 고쳐보겠씁니다 ㅠ
+    public RacingGame(InputDto inputDto) {
+        this.cars = create(inputDto.getCarNameList());
+        this.inputDto = inputDto;
+    }
+
+    // 테스트 코드를 위한 생성자
+    public RacingGame(List<Car> cars) {
+        this.cars = cars;
+    }
+
+    // 외부에서 tryCount를 매개변수로 받아야 더 나을지.. 지금 처럼 필드로 InputDto 받아서 사용하는 것이 나을지 모르겠습니다 흑흑
+    public void start() {
+        for (int i = 0; i < inputDto.getTryCount(); ++i) {
+            move(cars);
+        }
+    }
+
+    private void move(List<Car> cars) {
+        for (int i = 0; i < cars.size(); ++i) {
+            Car car = cars.get(i);
+            car.move(new RandomMovingStrategy());
+            ResultView.print(car);
+        }
+        System.out.println();
+    }
+
+    // 테스트 코드를 작성하기 위해서 public 으로 만들어둔.. 메소드 인데 괜찮을 지 모르겠습니다 ㅠ(기존 메소드를 그냥 public 하는게 나을까요..)
+    public List<Car> createTest(List<String> carNameList) {
+        return create(carNameList);
+    }
+
+    private List<Car> create(List<String> carNameList) {
+        List<Car> carList = new ArrayList<>();
+
+        for (int i = 0; i < carNameList.size(); ++i) {
+            carList.add(new Car(carNameList.get(i)));
+        }
+
+        return carList;
+    }
+
+    public String findWinner() {
+        int maxDistance = maxDistance();
+
+        return cars.stream()
+                .filter(car -> car.getPosition().equals(new Position(maxDistance)))
+                .map(Car::getCarName)
+                .collect(Collectors.joining(", "));
+    }
+
+    private int maxDistance() {
+        // getter 너무 많이 쓰는데 줄이는 방법 생각 ! !  (너무 난잡) => Refactor 필요 (하.. 떠오르는 생각은 getter 뿐.. 부족합니다..)
+        cars.sort((o1, o2) -> o2.getPosition().getPosition() - o1.getPosition().getPosition());
+        return cars.get(0).getPosition().getPosition();
```

Comment:
정렬을 위해 Position에서 getter를 사용해주셨고 getter에도 주석을 달아주신 이유가 여기였군요..!
우선 getter를 많이 사용하기에 줄여야 된다고 생각하시는 부분을 칭찬 드리고 싶네요 👍 
getter를 많이 사용하는 것은 내부 접근이 많이 일어난다는 뜻이고 많은 정보들이 노출되고 있다는 뜻으로
**캡슐화가 지켜지지 않고 있다는 신호**이기도 합니다.
캡슐화를 지키기 위해 많은 고민을 해주시는게 보이시네요 👍 

음.. getter를 지양하기 위해선 해당 객체에게 역으로 요청을 하여 요청처리를 해주도록 하면 어떨까요 :)?
현재 RacingGame에서 Position 객체의 값을 꺼내와서 정렬 기준을 세우고 있지만 역으로
Position에게 정렬 기준을 직접 세우도록 요청을 하거나 구현하면 될 것 같아요 :)
이것을 `객체에게 메세지를 보내 객체 스스로가 수행하도록 해라` 라고 표현할 수 있습니다.

다음과 같이 Position 객체가 Comparable을 구현하면 어떨까요 :)?
```suggestion
        Collections.sort(cars);
        return cars.get(0).getPosition().getPosition();
```
```java
public class Position  implements Comparable<Position> {
    @Override
    public int compareTo(Position other) {
        return Integer.compareTo(other.position, this.position);
    }
}

```

Reply:
오오 좋은 말씀이신 거 같습니다.. 저도 뭔가 어디선가 설계?에 문제가 있기에 저렇게 getter를 여러 번 쓰게 되는 거 같다고 생각했습니다 ㅠ 말씀해주신 것과 여러가지 저의 생각을 더해서 리팩터링 해보겠습니다 감사합니다 ㅎㅎ